pkg/e2e/infra/infra.go

```diff
@@ -196,77 +236,43 @@ var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
+		Expect(err).NotTo(HaveOccurred())
 
-		g.By("Getting the latest version of the node")
+		By("Getting the latest version of the node")
 		node, err = getNodeFromMachine(client, machine)
-		o.Expect(err).NotTo(o.HaveOccurred())
+		Expect(err).NotTo(HaveOccurred())
 
-		g.By("Setting back the original node taints")
+		By("Setting back the original node taints")
 		node.Spec.Taints = originalNodeTaints
 		err = client.Update(context.TODO(), node)
-		o.Expect(err).NotTo(o.HaveOccurred())
+		Expect(err).NotTo(HaveOccurred())
 	})
 
-	g.It("recover from deleted worker machines", func() {
-		var err error
-		client, err := e2e.LoadClient()
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		// Expect for cluster to cool down from previous tests
-		err = e2e.WaitUntilAllNodesAreReady(client)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By("checking initial cluster state")
-		initialClusterSize, err := getClusterSize(client)
-		err = waitForClusterSizeToBeHealthy(client, initialClusterSize)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By("getting worker node")
-		workerNodes, err := e2e.GetWorkerNodes(client)
-		o.Expect(err).NotTo(o.HaveOccurred())
-		o.Expect(workerNodes).ToNot(o.BeEmpty())
-
-		workerNode := &workerNodes[0]
-		workerMachine, err := e2e.GetMachineFromNode(client, workerNode)
-		o.Expect(err).NotTo(o.HaveOccurred())
-		g.By(fmt.Sprintf("deleting machine object %q", workerMachine.Name))
-		err = deleteMachine(client, workerMachine)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By(fmt.Sprintf("waiting for node object %q to go away", workerNode.Name))
-		nodeList := corev1.NodeList{}
-		o.Eventually(func() bool {
-			if err := client.List(context.TODO(), &nodeList); err != nil {
-				glog.Errorf("Error querying api for nodeList object: %v, retrying...", err)
-				return false
-			}
-			for _, n := range nodeList.Items {
-				if n.Name == workerNode.Name {
-					glog.Infof("Node %q still exists. Node conditions are: %v", workerNode.Name, workerNode.Status.Conditions)
-					return false
-				}
-			}
-			return true
-		}, e2e.WaitLong, 5*time.Second).Should(o.BeTrue())
+	It("recover from deleted worker machines", func() {
+		selector := machineSet.Spec.Selector
+		machines, err := e2e.GetMachines(client, &selector)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(machines).ToNot(BeEmpty())
+
+		machine := machines[0]
 
-		g.By(fmt.Sprintf("waiting for new node object to come up"))
-		err = waitForClusterSizeToBeHealthy(client, initialClusterSize)
-		o.Expect(err).NotTo(o.HaveOccurred())
+		By(fmt.Sprintf("deleting machine object %q", machine.Name))
+		err = deleteMachine(client, machine)
+		Expect(err).NotTo(HaveOccurred())
+
+		e2e.WaitForMachineSet(client, machineSet.GetName())
 	})
 
-	g.It("grow and decrease when scaling different machineSets simultaneously", func() {
-		client, err := e2e.LoadClient()
-		o.Expect(err).NotTo(o.HaveOccurred())
+	It("grow and decrease when scaling different machineSets simultaneously", func() {
```

Comment:
Looks like this test requires 2 different machinesets, I'll find a way to make it work